examples/permissioned-token/assets/permissions.py

```diff
@@ -0,0 +1,129 @@
+from pyteal import *
+
+def approval_program():
+    """
+    This smart contract is the permissions or rules smart contract.
+    If configured in controller, then this sc will ensure 2 rules are
+    followed during a token transfer:
+        a) receiver asset balance should be <=100 after transfer
+        b) both [from, to] accounts must be whitelisted
+    Permissions manager manages the contract permission.
+    It can be changed by the asset manager
+    """
+
+    true = Int(1)
+    max_tokens = Bytes("max_tokens") # maximum token transfer rule
+    whitelist_count = Bytes("whitelist_count") # whitelisted accounts counter
+    controller_app_id = Bytes("controller_app_id") # token's controller application index
+
+    on_deployment = Seq([
+	Assert(And(
+		Txn.application_args.length() == Int(1),
+
+		# Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+		# unless the contract is specifically involved ina rekeying operation.
+		Txn.rekey_to() == Global.zero_address()
+        )),
+
+        # Save max_tokens count in global state (default = 100, during ssc deploy)
+        App.globalPut(max_tokens, Int(100)),
+
+        # Initialize whitelisted accounts counter to 0
+        App.globalPut(whitelist_count, Int(0)),
+
+        # Save controller application index in global state
+        App.globalPut(controller_app_id, Btoi(Txn.application_args[0])),
+        Return(Int(1))
+    ])
+
+    # fetch permissions manager, token_id from controller's global state (using foreignApps)
+    permission_manager = App.globalGetEx(Int(1), Bytes("permissions_manager"))
+    token_id = App.globalGetEx(Int(1), Bytes("token_id"))
+
+    # add an account to be whitelisted (token transfer is rejected if account is not whitelisted)
+    add_whitelist = Seq([
+        permission_manager,
+        token_id,
+        Assert(And(
+		Txn.application_args.length() == Int(2),
+		Txn.accounts.length() == Int(1),
+		Txn.rekey_to() == Global.zero_address(),
```

Comment:
can we move it to a "common checks (2 lines above)

Reply:
moved rekey check to common var